### ISSUE TITLE
Open bookmark into Container

### ISSUE THREAD

**Issue by gwynnarth** [mozilla/multi-account-containers#323]
Currently it is possible to open a link that's on a page into a specific container by right-clicking and choosing appropriate "context" from dropdown menu, but you cannot do that with a bookmark, neither from Bookmark bar nor from Bookmarks menu.
It would be useful to be able to right-click a bookmark and open it into a container. Or might be even more useful if user could associate a particular bookmark with a particular Container so that it automatically opens in the right Container. So when I left-click "Facebook" bookmark it opens in "Facebook" container, when I left-click my Online banking website it automatically launches into "Finances" container.

**Comment by aleth** [comment]
Similarly, in the Bookmarks menu, "Open All in Tabs" could easily be extended to "Open All in... >" with a submenu of containers.

**Comment by jonathanKingston** [comment]
There is a platform bug for this as I'm not sure we can add this to the experiment but certainly can add it to the Nightly changes we will be doing: https://bugzilla.mozilla.org/show_bug.cgi?id=1334696

**Comment by aperkins81** [comment]
Until I saw this, I was nearly going to create an issue that adding a matching Tag to a bookmark does not auto-container bookmarks.  Glad to hear it's being looked at, would be a useful tool.

**Comment by biva** [comment]
In addition, it would be great that the behavior takes the keyboard shortcuts into consideration, as suggested in https://github.com/mozilla/testpilot-containers/issues/119

**Comment by klint** [comment]
This is definitely what I miss the most today with containers.
With the following default behaviour, IMHO: assuming you have a tab already opened in a certain container, left clicking from that tab on a bookmark that has no default container assigned (assuming that feature exists, of course ;)) should open the bookmark in the same container, either in the same tab or in a new tab, depending on the default setting for new tabs in Firefox pertaining to when new tabs are opened (see Tab Mix Plus settings).

**Comment by MurzNN** [comment]
@klint Can you describe how to configure open new tab in same container (in Tab Mix Plus settings)? For me all bookmarks opens in new tab with no container, even if current tab have a container. But if I open bookmark in private tab, it opens new tab in private mode too.

**Comment by klint** [comment]
@MurzNN Yes, that's the same for me. Sorry, my explanations were confusing. I have mentionned Tab Mix Plus just to say that my Firefox is configured to open any link or bookmark in a new tab, that's all.
But container-wise, it behaves exactly like you say. Which does not please me :)

**Comment by MurzNN** [comment]
Seems that [Private tab](https://addons.mozilla.org/en-US/firefox/addon/private-tab/) extension can successfully add menuitems into Bookmarks right-click menu, I see "Open in private tab" menu item after installing extension. Maybe you can lookup it's code and try to add menu item "Open bookmark into Container" via same way?

**Comment by anewuser** [comment]
@MurzNN  Private Tab not only has menu items, but lets you add `private:` before a URL to automatically open it in a private tab. I have dozens of bookmarks with keywords/custom searches set up that way.

**Comment by MurzNN** [comment]
Seems in this Firefox issue https://bugzilla.mozilla.org/show_bug.cgi?id=1370499 there are good progress with implementing access to Bookmarks menu via Webextensions API.

**Comment by klint** [comment]
I also see another usage of a opening a bookmark in a given container either by right-click option or by assigning a container to a bookmark, in the use-case of multiple identities for a same service.
Using containers to access Google under several accounts/containers, it would be possible to right-click on a generic "drive.google.com" bookmark to open it under the wanted account/container. Or it would be possible to have several bookmarks to the same URL "drive.google.com", each of them assigned to a specific account/container, and a simple left-click on the right bookmark would do the job.

**Comment by MichaelTunnell** [comment]
Being able to open bookmarks in containers would be wonderful but assigning containers directly to bookmarks would be absolutely amazing and solve a lot of headaches I currently have to deal with. I want to open a single website with multiple sessions which currently is a pain, if I could assign a container to a bookmark then I could accomplish this with so little effort that I would be amazed.

**Comment by MurzNN** [comment]
@MichaelTunnell For assigning containers directly to bookmarks we can register specific protocols for each container http://kb.mozillazine.org/Register_protocol

If we implement this, we can create bookmark links like `container-5:http://example.com/` or `container:work:http://another.example.com/page.html` that will opens in specific container. Now this is only idea for implementation.

**Comment by MichaelTunnell** [comment]
specific protocols would be a great option and something like `container:work:http://another.example.com/page.html` sounds like a great solution for this. I think this would be more useful overall to people who want to go this specific. Opening via the context menu would be nice for occasional usage but the ability to have relatively permanent way to open a bookmark in a particular container would be absolutely game changing.

**Comment by anewuser** [comment]
@MichaelTunnell, and once the "games changes" to you, it's awful knowing that it may be taken away from you. I've been doing that for years not to expose my saved cookies or pollute my history with URLs I don't want or need to be remembered. These are only a fraction of the bookmarks/custom searches I've mentioned:

![Bookmarks](https://user-images.githubusercontent.com/6733770/29128913-9810e628-7cfb-11e7-9b80-1ae9ffcc2c1f.png)

**Comment by MichaelTunnell** [comment]
yea I'm aware of Firefox doing things that removes key elements of my workflow. I used to be a huge user of Tab Groups and while these Container Tabs are somewhat similar and fundamentally better it lacks a huge benefit that Tab Groups had which was a more seamless UX.

However, I would hope this new approach too bookmarks and container tabs would be built into the addon rather than a 3rd party method as that would make sure it lasts as long as possible.

Overall the switch to webextensions is a good thing but it certainly has a lot of unfortunate addons that will be left in the wake of obsolescence.

**Comment by biva** [comment]
Any update on this one? Probably related to https://github.com/mozilla/multi-account-containers/issues/914

**Comment by MurzNN** [comment]
https://bugzilla.mozilla.org/show_bug.cgi?id=1370499

> Andy McKay [:andym] |  Comment 14 • 4 months ago
> Sounds like a good idea and no-one had any real objection in the meeting, so let's do this.

**Comment by klint** [comment]
Yes, as @MichaelTunnell said (sorry I don't know how to quote here!!): "Opening via the [bookmark] context menu would be nice for occasional usage but the ability to have relatively permanent way to open a bookmark in a particular container would be absolutely game changing."

I really miss this feature, now that I have move to FF57 totally. And I'm pleased with it.

**Comment by tradewatcher** [comment]
lol a simple feature request since more than 6 months for an important basic feature. xD

*upvote*

**Comment by davidclarke-au** [comment]
Have another vote from me, trying to migrate from Chrome, this is the big one stopping me at the moment.

Thanks

**Comment by MichaelTunnell** [comment]
@davidclarke-au how can you do this in Chrome?

**Comment by davidclarke-au** [comment]
@MichaelTunnell in Chrome, I have 2 profiles setup, one with my personal Gmail, and one with my Work GSuite account. This runs in 2 separate instances of Chrome, therefore, any new tab, link etc is always opened within that context.

I also tried about:profiles but couldn't get this to work.

**Comment by MurzNN** [comment]
Seems feature is successfully implemented in Firefox:
https://bugzilla.mozilla.org/show_bug.cgi?id=1370499
> Support WebExtensions bookmark context menus
RESOLVED FIXED in Firefox 59

**Comment by MichaelTunnell** [comment]
@davidclarke-au I see. That's kind of what I expected but I do not like the profile structure because I dont want a separate window always. It would still be good to have this separation though.

By the way, Chrome has separate instances for literally every tab and even more for each runtime stuff, not just profiles. That is why it uses up so much RAM when you have lot of tabs.

----------

@MurzNN while that is good . . . context menu is almost not really important to me when compared to assigning containers directly to bookmarks with specific protocols  . . . I want that so freaking much.

**Comment by MurzNN** [comment]
"assigning containers directly to bookmarks" is another feature, that also will be good, but current issue is for menu items. Let's create separate feature request about assigning containers directly to bookmarks?

**Comment by davidclarke-au** [comment]
@MichaelTunnell spot on and I agree, it's high memory usage was the catalyst for me to see if I could move to Firefox, so far so good, just a few small gotcha's.

**Comment by MurzNN** [comment]
Mozilla implement function for modify bookmarks menu: https://bugzilla.mozilla.org/show_bug.cgi?id=1370499#c32 - can we add this feature now?

**Comment by groovecoder** [comment]
Yes, looks like it will be possible in 59. Labeling help wanted and context menu.

**Comment by oueV** [comment]
> https://github.com/mozilla/multi-account-containers/issues/323#issuecomment-358548483

Seems like implementation will finally be possible as of **March 13th** on FF 59 release's launch &rlm; src: [bugzilla1370499#c34](https://bugzilla.mozilla.org/show_bug.cgi?id=1370499#c34)

**Comment by klint** [comment]
Hi 
Is there any update on the 2 requests here? Or are there known addons that fullfil the needs?
- open a bookmark to a specific container by right clicking
- assign a container to a bookmark to always open it to that container?
Thank you :)

**Comment by sheam** [comment]
This is something I am _really_ missing from Chrome. Chrome allowed you to manage your bookmarks per container which was great.

**Comment by mitchp** [comment]
Please. Please. Please.  Add "open bookmark in container" to context menu when over a bookmark.  Then, submenu with container list so you can just click a container plus an "add new container" in that sub-context menu.

I have Gmail as a bookmark, but I want to open it in "mystuff" container sometimes or "work" or "some client" (different gmail accounts)...  I don't want 10 gmail bookmarks, each assigned to a different container.

**Comment by mitchp** [comment]
Thanks!

**Comment by Rob--W** [comment]
I've created an add-on that adds a context menu item to the bookmarks menu that offers the ability to open a bookmark in a container tab. If a bookmark folder is clicked, all links in this folder will be opened in container tabs (i.e. the behavior requested in #1251).

Install from: https://addons.mozilla.org/en-US/firefox/addon/open-bookmark-in-container-tab/
Source code / issue tracker: https://github.com/Rob--W/bookmark-container-tab

**Comment by K-Langoso** [comment]
Any news on this front?

Thanks!

**Comment by Millichrome** [comment]
I am currently using Rob--W's extension (Thanks), would love to have this integrated in the containers feature.

**Comment by thelaith** [comment]
Any news?

**Comment by MichaelTunnell** [comment]
@xboss182 not possible, sadly.

@thelaith not really. I don't expect this to ever happen at this point by the official addon since it has been almost 2 years since requested. Our best bet is for a separate addon to be made but that's not happening yet either. @Rob--W addon is nice but doesn't really accomplish the overall goal.

**Comment by xboss182** [comment]
@MichaelTunnell thanks mate

**Comment by caltheon** [comment]
Someone made an extension to add right-click bookmark->open in container->Container

https://addons.mozilla.org/en-US/firefox/addon/open-bookmark-in-container-tab/

**Comment by GodKratos** [comment]
I've finally found a work around solution that makes this behaviour work as desired thanks to the post [here](https://www.reddit.com/r/firefox/comments/8u28z6/how_to_make_a_bookmark_that_always_open_in_a/)

Using the [Redirector](https://addons.mozilla.org/en-US/firefox/addon/redirector/) addon I made two rules as follows:
**Rule 1**
- Include Pattern: *#redirect-*
- Redirect to: http://www.$2.myredirect/$1
- Pattern type: Wildcard

**Rule 2**
- Include Pattern: http://www.*.myredirect/*
- Redirect to: $2
- Pattern type: Wildcard

![image](https://user-images.githubusercontent.com/692616/58224702-daac9d00-7d72-11e9-81a6-f62494b1d8c6.png)

What this means is that I can add "#redirect-containername" to the end of any bookmark URL and then configure the URL "http://www.containername.myredirect" to auto open in the required container and when you click the bookmark it will switch to the redirect site, open in the container, then redirect back to the bookmark URL.

Works a treat and can be used for redirecting to any container you like.

**Comment by MurzNN** [comment]
@GodKratos, thanks for very good workaround!

**Comment by caltheon** [comment]
@GodKratos 

I follow all of it until the part where you add http://www.containername.myredirect to auto open in the required container.  How do you manage that?  Since it isn't a real URL, I can't do it the normal way of going to the site and choosing open in this container, it just gets redirected to ISP search page.  I may be blind, but I don't see any way to configure the website list explicitly in the container plugin either.

**Comment by MichaelTunnell** [comment]
@GodKratos that is a great solution! it is convoluted and has a lot of setup but once it is done it works perfectly so thank you for sharing this solution!

@caltheon ISP redirects are awful. There are a few options though. 1.) disable internet like you did. 2.) change your DNS provider. 3.) use a separate extension ([Containerise](https://addons.mozilla.org/en-US/firefox/addon/containerise/?src=search)) to handle that

**Comment by GodKratos** [comment]
@caltheon I just browsed to the redirect URL in my address bar and it just fails to the load the site but still allows me to add it to the container. Remember to enter in the URL including the http:// part into the address bar and not the search bar. If you don't add the http:// it will try to do a search but it also sounds like your ISP does some search thing for unknown URLs?

**Comment by rafael-gssa** [comment]
Hi all, I've created an extension that makes the workaround posted by @GodKratos easier / faster to manage. Thanks for the workaround, by the way!

Install here: https://addons.mozilla.org/en-US/firefox/addon/container-bookmarks/
Source code: https://github.com/rafael-gssa/container-bookmarks

**Comment by olympus123** [comment]
I've found this new add-on
Amaaaziiing, it works for bookmarks and it even creates the container if it doesn't exist!
It doesn't even need Multi-Accounts add-on or Containerise to work...

Open external links in a container
This extension enables support for opening links in specific containers using custom protocol handler. It works for terminal, OS shortcuts and regular HTML pages.

https://addons.mozilla.org/en-US/firefox/addon/open-url-in-container/

ext+container:name=MyContainer&color=orange&icon=fruit&url=https://mozilla.org

**Comment by rafaelgssa** [comment]
Wow, hadn't even thought of protocol handlers, amazing. Only problem for bookmarks is the icons not appearing, but for everything else, specially for opening from the terminal, it's great!

**Comment by olympus123** [comment]
> Wow, hadn't even thought of protocol handlers, amazing. Only problem for bookmarks is the icons not appearing, but for everything else, specially for opening from the terminal, it's great!

Yes, firefox can't get the favicons.
I personally use Humble New Tab Page, which shows my bookmarks in new tab page.
I don't use bookmarks toolbar because Firefox sync can't sync favicons and in a new install I would have to open all bookmarks to save their new icons.
Humble New Tab Page can get their favicons automatically if I select to get them from faviconkit.com
https://addons.mozilla.org/en-US/firefox/addon/humble-new-tab/

**Comment by rafaelgssa** [comment]
> > Wow, hadn't even thought of protocol handlers, amazing. Only problem for bookmarks is the icons not appearing, but for everything else, specially for opening from the terminal, it's great!
> 
> Yes, firefox can't get the favicons.
> I personally use Humble New Tab Page, which shows my bookmarks in new tab page.
> I don't use bookmarks toolbar because Firefox sync can't sync favicons and in a new install I would have to open all bookmarks to save their favicons.
> Humble New Tab Page can get their favicons automatically if I select to get them from faviconkit.com
> https://addons.mozilla.org/en-US/firefox/addon/humble-new-tab/

Thanks for the suggestion, but it doesn't look good for me. I have a ton of bookmarks, so I remove their names so that I only see icons in my bookmarks bar, seeing them in a column is not practical. I'll just keep using the extension I made for bookmarks and the one you posted for everything else.

**Comment by MichaelTunnell** [comment]
@olympus123 great find, thank you for sharing! This was the ideal method that @MurzNN and I were talking about 2 years ago as a solution to this issue. I have installed it and tested it and it works beautifully!

It is true that the favicons don't work anymore due to this method but with that being the only downside to using this vs the Redirector method of extra setup time, I think this is a wonderful solution.

I really love the fact this works in a terminal and if you use a container name that doesn't exist it will create the container for you so you could use this as a way to sync containers in future installs. Import your bookmarks and just launch some bookmarks to create the containers. I think this is something I will make a video about. :D

**Comment by olympus123** [comment]
> > Yes, firefox can't get the favicons.
> > I personally use Humble New Tab Page, which shows my bookmarks in new tab page.
> > I don't use bookmarks toolbar because Firefox sync can't sync favicons and in a new install I would have to open all bookmarks to save their favicons.
> > Humble New Tab Page can get their favicons automatically if I select to get them from faviconkit.com
> > https://addons.mozilla.org/en-US/firefox/addon/humble-new-tab/
> 
> Thanks for the suggestion, but it doesn't look good for me. I have a ton of bookmarks, so I remove their names so that I only see icons in my bookmarks bar, seeing them in a column is not practical. I'll just keep using the extension I made for bookmarks and the one you posted for everything else.


https://fitzcarraldoblog.wordpress.com/2018/10/13/how-to-create-missing-favicons-for-your-bookmarks-in-firefox-quantum-for-linux/

I tried to change the bookmark favicon with the above method
It works, no need for an add-on, it works withuserChrome.css
It does change the favicon, but it needs the name of the bookmark.
Maybe you can modify it a little to check the url instead of the name of the bookmark.

**Comment by kendallcorner** [comment]
Hi everyone,

I'm willing to look at this item. It looks like this issue is for adding the containers options to the bookmark menu, and a separate issue was opened for assigning a bookmark a container. So I'm only going to look at the bookmark context menu part.

@maxxcrawford can you assign this to me?

**Comment by maxxcrawford** [comment]
@kendallcorner Done!

**Comment by MichaelTunnell** [comment]
> and a separate issue was opened for assigning a bookmark a container. So I'm only going to look at the bookmark context menu part.

**All of the separate issues** for making bookmarks open directly with a container without a context menu **were closed** because project members/maintainers claimed it was a duplicate of this one.

Many many people explained how they are similar but also DIFFERENT requests. In my opinion, the context menu is the least valuable piece because having multiple bookmarks of the same website automatically open to a particular container is much more powerful. But I digress . . .

With you taking on only one task of this thread proves a couple things:
1. it proves that the project just care about improving the container tabs system . . . it felt like it was just in a stale state.

2. that they are DIFFERENT things. So hopefully, now with you choosing to address only one piece it will make it so the other threads can be reopened and not ignored.

so once again, these issues [#1](https://github.com/mozilla/multi-account-containers/issues/854) and [#2](https://github.com/mozilla/multi-account-containers/issues/1142) are not duplicates of this thread . . . they are duplicates of each other but they are only repeated because they are NOT duplicates of this one.

**Comment by groovecoder** [comment]
With #1537, our next release **starts** to address container-bookmark integration & UX. As @MichaelTunnell points out, it does not solve **ALL** of the container-bookmark UX requests in this issue, or in #854 #1142 and #1443.

Container-bookmark UX is a big project, and some of the solutions will require changes in Firefox itself. Any changes here could introduce tricky bugs both in technical integration and in UX flows.

We will soon on-board an [Outreachy](https://www.outreachy.org/) intern dedicated to work on Multi-Account Container issues like this one. Multi-Account Container users are a passionate group of advanced power-users with strong opinions about browser tab UI & workflows.

We ask for your patience as we start to explore this advanced feature space, and please be nice to the contributors who work on Multi-Account Containers. Thank you awesome Firefox users!

**Comment by MichaelTunnell** [comment]
to be clear, I am very much thankful to @kendallcorner for taking on this task. I commented above just to make sure that this task wasn't completed, closed and the other piece be lost. Due to all the other threads being closed in error, I felt it might be necessary to clarify that they were mistakenly closed and still a very sought after feature.

with that said, @groovecoder that all sounds great with bringing in work back into Containers extension. You are right that many power-users use this feature and it might be something worth considering in terms of workflow and how it is addressed.

I am excited to see further expansion on the Containers system. It's my #1 reason that I love Firefox these days. :D

**Comment by piecevcake** [comment]
Please see larger context which also provide an interim workaround to avoid privacy breaches at https://github.com/mozilla/multi-account-containers/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+import+tabs

**Comment by kendallcorner** [comment]
Check out the update released today. You can turn on Bookmark context menus in the add-on permissions.

**Comment by kendallcorner** [comment]
Possibly a controversial question, but since the latest release implements the bookmark context menu (you have to turn it on in the add-on settings) and #854 is open to cover assigning a container to a bookmark ... can I close this issue, or should I leave it open?

**Comment by StefanSchoof** [comment]
I have a site where I need to work with different user. In my mind a good solution would be two bookmarks:
 - System User A
 - System User B
Then I could type in the Bar "System" and select with the arrow keys the right Bookmark and depending of the Bookmark it opens in a other container.
So for me is reason I subscribe here not solved.

**Comment by MichaelTunnell** [comment]
> Possibly a controversial question, but since the latest release implements the bookmark context menu (you have to turn it on in the add-on settings) and #854 is open to cover assigning a container to a bookmark ... can I close this issue, or should I leave it open?

Because the development team flat out *refused* to allow separating threads on the issue of assigning containers to a bookmark and insisted that this thread covered both topics even though the mention was an after thought . . . then I think this is most certainly not a closable issue until both tasks are complete.

I am thrilled that these are being done but since they insisted that this is a 2 part thread then both parts need to be done before closing.

**Comment by piecevcake** [comment]
How does open bookmark in container addon do it? ? yu were able to open a folder of bookmarks in container chosen from context menu

This doesnt address the issue of other imported pages- onetab group, tree tabs group import?

**Comment by rafaelgssa** [comment]
> > > Yes, firefox can't get the favicons.
> > > I personally use Humble New Tab Page, which shows my bookmarks in new tab page.
> > > I don't use bookmarks toolbar because Firefox sync can't sync favicons and in a new install I would have to open all bookmarks to save their favicons.
> > > Humble New Tab Page can get their favicons automatically if I select to get them from faviconkit.com
> > > https://addons.mozilla.org/en-US/firefox/addon/humble-new-tab/
> > 
> > 
> > Thanks for the suggestion, but it doesn't look good for me. I have a ton of bookmarks, so I remove their names so that I only see icons in my bookmarks bar, seeing them in a column is not practical. I'll just keep using the extension I made for bookmarks and the one you posted for everything else.
> 
> I tried to change the bookmark favicon with userChrome.css
> It works, no need for an add-on, it works with userChrome.css
> It does change the favicon, but it needs the name of the bookmark.
> Maybe you can modify it a little to check the url instead of the name of the bookmark.
> Have a look at
> 
> https://fitzcarraldoblog.wordpress.com/2018/10/13/how-to-create-missing-favicons-for-your-bookmarks-in-firefox-quantum-for-linux/
> 
> EDIT
> 
> I've made some tests.
> 
> I created a YouTube bookmark which opens in YouTube container
> 
> YouTube
> ext+container:name=YouTube&color=red&url=https://www.youtube.com
> 
> My userChrome.css, it adds the favicon and removes the text from it (the way you want it to be)
> 
> @namespace url("http://www.mozilla.org/keymaster/gatekeeper/there.is.only.xul");
> 
> .bookmark-item[label="YouTube"] image {
> width:0!important;
> height:0!important;
> padding: 0 0 16px 16px !important;
> background:url('https://i.imgur.com/lEpltc3.png')!important;
> }
> 
> .bookmark-item[label="YouTube"] > .toolbarbutton-text {
> display: none !important;
> }

Thank you, this works perfectly! For those who want to target by URL, just do `.bookmark-item[image*="youtube.com"]`, for example.

**Comment by maxxcrawford** [comment]
There is a setting now in the add-on preferences to enable this functionality! 

Go to `about:addons` and select Multi-account Containers: 

<img width="593" alt="image" src="https://user-images.githubusercontent.com/2692333/87975041-ac1e7500-ca90-11ea-8b60-e1b389ed0e9d.png">

---

Right+click/option-click on a Bookmark or bookmark folder and go to "Open in Container Tab…"

<img width="592" alt="image" src="https://user-images.githubusercontent.com/2692333/87975280-07506780-ca91-11ea-9de7-b9622646170a.png">

**Comment by rafaelgssa** [comment]
@maxxcrawford That setting doesn't fix the other issue that was brought up, as has already been mentioned just comments above yours. We want a functionality to **assign a container to a bookmark**, so that the bookmark always opens in that container.

**Comment by mvdan** [comment]
Friendly ping about what @rafaelgssa said. Should we open a new issue if this one will just remain closed?

**Comment by davidclarke-au** [comment]
Why doesn't the following solve this issue? 

Is it where the domain is the same and you want to pin it to the bookmark itself, not the TLD? 

![image](https://user-images.githubusercontent.com/28612479/88997454-1c4fb500-d333-11ea-91d3-c9eb7222932e.png)

**Comment by caltheon** [comment]
Exactly.  Having multiple account logins within a single domain this doesn't work.  For instance, having a container on github for multiple users and being able to log into each one at the same time in different tabs without having to manually assign a container for each instance.  Lots of sites will overwrite your login session if you log in to two accounts in the same context/container

**Comment by MichaelTunnell** [comment]
@davidclarke-au as @caltheon stated, the reason this doesn't address the issue is because if the user wants to login to multiple instances of the same website they can't use this. If you could associate a container to a bookmark then you could have all of the different accounts organized in a folder and then right click folder, select "Open All in Tabs" and from here open each account in a new tab associated to their respective containers all automatically and simultaneously. This feature would be a massive game changer and way more useful than the other related requests once implemented, in my opinion. (technically possible via an addon but should be built in for reliability as it is a fundamental value)

As for why **this thread should never be closed until** this association to bookmarks feature is implemented is because it took the proponents of this feature 3 years to convince the repo maintainers that it is indeed a separate request. The maintainers insisted that this thread was all that was needed to cover both requests even though the feature we want is a secondary off-handed throwaway on this request.

> Or might be even more useful if user could associate a particular bookmark with a particular Container so that it automatically opens in the right Container.

This one sentence stated by @gwynnarth as a secondary idea to the main point of the thread inadvertently locked the feature request we wanted to this thread because the repo maintainers **insisted** that it was the same thing. We kept trying to explain that all they were doing was burying the request inside another request. I predicted that someone would come around and pick up only the context menu request and ignore the other more useful piece because it was buried. This is exactly what happened.

[Every time](https://github.com/mozilla/multi-account-containers/issues/854) one of us [created a thread](https://github.com/mozilla/multi-account-containers/issues/1142) to discuss this specific feature request they closed them all. They refused for 3 years to listen to us that it was a different thing and therefore needed separate thread to have discussion and to test engagement of interest.

So this thread should be forcibly kept open until #854 is completed because we had to wait years for that thread to even be *considered* separate so this thread should not be closed separately. This thread should only ever be closed as complete when #854 is closed as complete.

**Comment by gwynnarth** [comment]
In hindsight (3 years gives you a hell of a hindsight indeed...) I must say that combining two very different (related but still different) ideas into one issue was a mistake on my part. I have naively hoped that maintainers will either treat them as inseparable and swiftly implement both ideas, or will create a different issue to track the secondary idea (for which I didn't have a clear usage example at the time).

From the experience of the last 3 years I can definitely say that this secondary idea is a better one and would be useful to a lot more people (myself included). And it makes me sad to see that various efforts to underline that these are indeed two completely different use cases were thwarted by repo maintainers for no good reason - after all it didn't result in both of these ideas being implemented here. Fighting ferociously to declare all those other issues as "duplicates" was just a waste of everyone's time 😔

TL;DR: #Sorry-854 and don't close this issue until we get "permanently assign a bookmark to a container" implemented.

**Comment by MarkCallow** [comment]
I just started using containers. Since the need for a bookmark to open in a particular container is so screamingly obvious I thought that if I bookmarked a page that is open in a container that all future opens of that bookmark would open it in the same container. Silly me. Not only that but there is NO way in the base Containers functionality, as far as I can see, to even get a URL to always open in a particular container. In my view that makes Containers completely useless. I had to install the multi-account containers add-on to get that ability. And the UI for it is an awful 2 step process. Step 1: left click on the multi-account containers add-on icon and select "Always open in ..." at the conclusion of which it says "success" or something similar. But then when I click the bookmark I discover the authors have a different interpretation of "success" from me. I get the dialog shown in https://github.com/mozilla/multi-account-containers/issues/323#issuecomment-666896405 asking me to confirm the decision I just made.

Given that this issue has been open for almost 4 years, I'm not holding my breath for any improvements.

**Comment by tylerszabo** [comment]
@MarkCallow have you tried using the `ext+container:name=Profile1&url=https://www.example.com/` `ext+container:name=Profile2&url=https://www.example.com/` - it's not ideal but it does allow you to have bookmarks for the same URL and have them open in different containers and that can be used to manage multiple logged-in accounts for services that don't natively support that.

Selecting "always open in container" will override the bookmark, though (which makes sense for "always"). I don't know if there's a way to have a "default to this container" but I think that would be a separate issue.

It would be nice to have bookmarks support containers directly and not need to use the `ext+container:` scheme.

**Comment by grahamperrin** [comment]
https://github.com/mozilla/multi-account-containers/issues/323#issuecomment-666901198

> multiple account logins within a single domain 

For this, I use a complementary extension. See https://github.com/mozilla/multi-account-containers/issues/854#issuecomment-751410530

**Comment by grahamperrin** [comment]
https://github.com/mozilla/multi-account-containers/issues/854#issuecomment-751410530

> # Open bookmark into Container

This (the opening post) is, essentially, resolved as pictured at https://github.com/mozilla/multi-account-containers/issues/323#issuecomment-661273776

> Open Bookmark in Container Tab ▶ …

Continuing to discuss #854 under #323 serves no useful purpose; people can simply discuss #854 under #854 

Given the resolution, I respectfully suggest closing this issue. Thanks

**Comment by MichaelTunnell** [comment]
@grahamperrin 
> Given the resolution, I respectfully suggest closing this issue. Thanks

I respectfully disagree and so does the original poster of this issue. The effort to make #854 to be accepted as an issue took 3 years for no reason because maintainers repeatedly closed it as a duplicate of this issue because they refused to listen to why it was different. Now that it is acknowledged that it is a different issue this issue should only be closed when #854 is due to the 3 year delay they forced on it.

If you want more info as to why here is a much longer version of the explanation: [issuecomment-667159313](https://github.com/mozilla/multi-account-containers/issues/323#issuecomment-667159313)

**Comment by grahamperrin** [comment]
Focusing on the technical: 

* #323 here is suitably titled
* we have links from unresolved #854 to resolved issue #323
* #854 is suitably titled

**Comment by grahamperrin** [comment]
> This isn't so trivial to implement,

It's implemented. Did you post to the wrong issue?

**Comment by wsdfhjxc** [comment]
Only the context menu part is implemented. Quoting OP:

>[...] might be even more useful if user could associate a particular bookmark with a particular Container so that it automatically opens in the right Container [...]

And then the big chunk of the thread basically revolves around that idea. It's not implemented. If the scope of this issue was only about the context menu, the issue should have been closed long time ago, and yet it isn't. The thread is still open, and active supporters of the idea [don't even want it to be closed](https://github.com/mozilla/multi-account-containers/issues/323#issuecomment-586104455), until said automatic feature is implemented.

You're right that there is an another issue with a more specific title, so my post could fit better there. However, I'll dare to say, that doesn't make it invalid here, given all of the above. If what you're after is that there should be no more noise in this particular thread, then okay, I'm sorry for that, and I won't post here again. But really, in such case, keeping this issue open doesn't help anyone...

**Comment by caltheon** [comment]
If the issue is the bookmark handler cannot support the functionality requested, and I assume getting that changed is probably a non-starter, would it make sense to change this to a feature request to add bookmarks to the container plugin itself to bypass the issue?

**Comment by grahamperrin** [comment]
> the context menu part 

That was this issue, which is implemented. 

https://github.com/mozilla/multi-account-containers/issues/854 is relevant to what's not implemented.

**Comment by JaumeNoguera** [comment]
This would be very useful to me. It would allow to open a bookmark directly to a specific container. Further, it would allow to assign the same website to different containers, using different bookmarks, then open them with no extra clicks needed. When you need multiple bookmarked containers for the same website (as I do need), this bypasses the problem "open always in container" , which effectively blocks using multiple containers in the same website.

**Comment by dannycolin** [comment]
Hey all,

I'm moving this feature request to our new feedback platform [Mozilla Connect](https://connect.mozilla.org)
so it gets more visibility.

On this platform if an idea has received a great deal of support (votes, comments)
it will be tagged as "Trending Idea" and the Mozilla team is going review it for
potential adoption.

Please upvote and comment on the following idea:

https://connect.mozilla.org/t5/ideas/container-preference-in-bookmarks/idi-p/3857
